### PR TITLE
Connect DB with IPAddress

### PR DESCRIPTION
**Pull request by mohitrawat789**

### CODE REVIEW

File: Parking.Common/ConfigurationReader.cs [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
 
 namespace Parking.Common
 {
-    public class ConfigurationReader
+    public sealed class ConfigurationReader
     {
-        private string ConfigFilePath { get; set; }
-
-        public ConfigurationReader(string folder, string file)
+        private static ConfigurationReader instance = null;
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Common/ConfigurationReader.cs]: Fixed

File: Parking.Common/ConfigurationReader.cs [hunk 2]
```diff
@@ -1,36 +1,51 @@
...
+        private string ConfigFilePath = Path.Combine(Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().GetName().CodeBase.Substring(8)), "DeviceConfig.json");
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Common/ConfigurationReader.cs]: Done

File: Parking.Common/Logger.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.IO;
+
+namespace Parking.Common
+{
+    public class FileLogger
+    {
+        private static string filePath = Path.Combine(Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().GetName().CodeBase.Substring(8)), "logs.txt");
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Common/Logger.cs]: done

File: Parking.Common/Logger.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        public static void Log(string message)
+        {
+            using (StreamWriter streamWriter = new StreamWriter(filePath))
+            {
+                try
+                {
+                    streamWriter.WriteLine(DateTime.Now + " :: " + message);
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Common/Logger.cs]: done

File: Parking.SerialPort/SerialPortCommunicate.cs [hunk 5]
```diff
@@ -57,20 +57,8 @@ public void RegisterVehicleEntryCallBack(Action<string> vehicleEntryCallback)
             }
             catch (Exception e)
             {
-                Console.WriteLine(e);
+                FileLogger.Log($"No callback assigned for Vehicle Entry : {e.Message}");
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.SerialPort/SerialPortCommunicate.cs]: Done

File: Parking/Forms/Settings.cs [hunk 6]
```diff
@@ -22,24 +22,30 @@ public partial class Settings : Form
...
             InitializeComponent();
-            parkingDatabaseFactory = new ParkingDatabaseFactory();
-            var configrationReader = new ConfigurationReader(Application.ExecutablePath, @"DeviceConfig.json");
-            tdSetting = configrationReader.Load();
+            parkingDatabaseFactory = new ParkingDatabaseFactory();            
+            tdSetting = ConfigurationReader.Instance.GetConfigurationSettings();
 
-            if (tdSetting.DeviceId == null)
+            if (tdSetting.TDClientDeviceId == null)
                 MessageBox.Show("Problem Loading Configuration Information");
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking/Forms/Settings.cs]: Done

**Comment by ankitit2015** [comment]
Have you re push the changes ?

do push -f <remote-name> <branch name>

File: Parking/Forms/Settings.cs [hunk 7]
```diff
@@ -69,22 +75,28 @@ private void LoadSettings()
...
+
+                var result = serialPortCommunicate.Connect(portName, BAUD_RATE, PARITY, DATA_BITS, STOP_BITS);
+                if (result)
+                {
+                    lblSettingStatus.Text = "Port " + portName + " connected successfully";
+                    lblSettingStatus.ForeColor = Color.Green;
+                }
+                else
+                {
+                    lblSettingStatus.Text = "Problem connecting to Port " + portName;
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking/Forms/Settings.cs]: Done

File: Parking/bin/Debug/logs.txt [hunk 8]
```diff
@@ -0,0 +1,2 @@
+Ticket Processing Failed as : Cannot insert the value NULL into column 'ParkingCharge', table 'db_Parking.dbo.tbl_parking'; column does not allow nulls. INSERT fails.
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking/bin/Debug/logs.txt]: Done

File: Parking/Forms/VehicleEntry.cs [hunk 9]
```diff
@@ -26,57 +25,71 @@ public VehicleEntry()
...
             InitializeComponent();
             parkingDatabaseFactory = new ParkingDatabaseFactory();
-            var configrationReader = new ConfigurationReader(Application.ExecutablePath, @"DeviceConfig.json");
-            tdSetting = configrationReader.Load();
 
-            if (tdSetting.DeviceId == null)
+            tdSetting = ConfigurationReader.Instance.GetConfigurationSettings();
+
+            if (tdSetting.TDClientDeviceId == null)
                 MessageBox.Show("Problem Loading Device Configuration");
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking/Forms/VehicleEntry.cs]: Done

File: Parking.Database/CommandFactory/ParkingDatabaseFactory.cs [hunk 10]
```diff
@@ -78,13 +93,13 @@ public DataRow GetMasterSettings()
...
             {
-                Console.WriteLine(exception);
+                FileLogger.Log($"TDClient Master settings could not be loaded successfully from database as : {exception.Message}");
                 throw;
             }
 
         }
 
-        public Ticket SaveVehicleEntry(string deviceId, string vehicleNumber, int vehicleType)
+        public Ticket SaveVehicleEntry(string deviceId, string vehicleNumber, int vehicleType, string parkingCharge, string MPSDeviceId)
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Database/CommandFactory/ParkingDatabaseFactory.cs]: Done

**Comment by mohitrawat789** [comment]
Fixed all comments and have pushed my changes again.

**Comment by ankitit2015** [comment]
Merged

**Pull request merged**